tests/unittests/tests-hashes/tests-hashes-keccak.c

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (C) 2017 Adrian Herrmann <adrian.herrmann@fu-berlin.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     unittests
+ * @{
+ *
+ * @file
+ * @brief       Test cases for the Keccak hashing package
+ *
+ * @author      Adrian Herrmann <adrian.herrmann@fu-berlin.de>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdint.h>
+
+#include "KeccakHash800.h"
+#include "KeccakHash1600.h"
+#include "embUnit.h"
+
+Keccak800Hash_instance* hash800Instance;
+Keccak1600Hash_instance* hash1600Instance;
+
+const BitLength databitlen = 80;
+const char datastring[] = "0123456789";
+const BitSequence* data = (unsigned char*) datastring;
+
+const char hashvalstring800[] =
+            "a6305bbe48f1f8d2c58dfd9731974fe85321c09cba8a944b0635a9ba07443324";
+BitSequence* hashval800;
+
+const char hashvalstring1600[] =
+            "8f8eaad16cbf8722a2165b660d47fcfd8496a41c611da758f3bb70f809f01ee3";
+BitSequence* hashval1600;
+
+static void setUp(void)
+{
+    /* Initialize */
+    hash800Instance = malloc(sizeof(Keccak800Hash_instance));
+    hashval800 = malloc(32);
+
+    hash1600Instance = malloc(sizeof(Keccak1600Hash_instance));
+    hashval1600 = malloc(32);
+}
+
+static void tearDown(void)
+{
+    /* Finalize */
+    free(hash800Instance);
+    free(hashval800);
+
+    free(hash1600Instance);
+    free(hashval1600);
+}
+
+static void test_keccak800(void)
+{
+    /* Testing hash initialization */
+    Keccak800Hash_256_initialize(hash800Instance);
+
+    /* Testing hash update */
+    Hash800Return return_update = Keccak800Hash_update(hash800Instance, data, databitlen);
```

Comment:
Typically in RIOT we use length in bytes (not in bits).
So here I suggest to use a wrapper Keccak800Hash_256_update that takes sizeof(data) as last parameter, instead of length in bits.

Reply:
@emmanuelsearch I'm not sure if this might be required by the hash function. For IP prefix lengths e.g. we also use bits instead of bytes ;-)

Reply:
mmmh but do we hash prefix lengths? We hash other stuff, that is generally measured in bytes, right?

Reply:
I didn't talk about IPv6 prefix lengths in general. That was just an example. What I wanted to say was: If the algorithm calls for it, the length should be given in bits.

Reply:
if I'm not mistaken the sponge-absorb step (the primitive called by hash-update) expects a number of bits that is divisible by 8...

Reply:
True... ^^ Then forget what I said.

Reply:
The implementation explicitly allows inputting data with a length that is not a multiple of 8 (but only in the last call of the update function) so I feel we should leave this as it is.